railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -159,6 +159,8 @@ def config_when_updating
     end
 
     def master_key
+      return if options[:pretend]
```

Comment:
Usually such options are called `--dry-run`. May be out of context here, but throwing in the suggestion.